packages/next-intl/test/react-server/index.test.tsx

```diff
@@ -57,4 +71,22 @@ describe('performance', () => {
     // Render 3: Recursive call that renders synchronously as well
     expect(renderCount).toBe(3);
   });
+
+  it('shares message format cache between useTranslations and getTranslations', async () => {
+    // First invocation
+    useTranslations('Component');
+    const firstCallCache =
+      vi.mocked(createTranslator).mock.calls[0][0].messageFormatCache;
```

Comment:
I think the test strikes a good balance by testing the instance of the `messageFormatCache`. Actually rendering a translation will be a bit more involved. From what I've seen async components can't be rendered in React 18.2, we'd have to upgrade to canary in the dev dependencies of `next-intl` (which seems to require other changes).

So tldr;—yep, I think this test is a good idea!